src/data/structure/UserProfile.java

```diff
@@ -57,16 +57,21 @@ public String getId() {
      * @param preferenceId
      * @return added preference
      */
-    public Preference addPreference(PreferenceCategory category, String preferenceId) {
+    public Preference addPreference(PreferenceCategory category, String preferenceId, int popularity,
```

Comment:
I very strongly oppose having these additional parameters in UserProfile.addPreference(). The reason is that, when a user adds a preference, they know nothing about its popularity or correlations, nor should they need to. The user is simply supposed to know WHICH preferences it has. That's why it previously had only the category and ID. If this is just to accommodate a change to the Preference constructor, then Preference should instead be enhanced to surface a constructor that provides default values for the other fields, as I'm pretty sure it did already.